web/server/codechecker_server/server.py

```diff
@@ -745,9 +745,17 @@ def __init__(self,
                 LOG.info("Initiating SSL. Server listening on secure socket.")
                 LOG.debug("Using cert file: %s", ssl_cert_file)
                 LOG.debug("Using key file: %s", ssl_key_file)
-                self.socket = ssl.wrap_socket(self.socket, server_side=True,
-                                              keyfile=ssl_key_file,
-                                              certfile=ssl_cert_file)
+                ssl_context = ssl.SSLContext(ssl.PROTOCOL_TLS_SERVER)
+                ssl_context.load_cert_chain(certfile=ssl_cert_file,
+                                            keyfile=ssl_key_file)
+                ssl_context.minimum_version = ssl.TLSVersion.TLSv1_2
```

Comment:
This option was introduced in `Python 3.7`, so with this change we will not support `Python 3.6` anymore.

In the source code couple of places we still mention/require python 3.6:

https://github.com/Ericsson/codechecker/blob/9004c1020fb9089c76b7454629b6d3fa96483e07/setup.py#L174
https://github.com/Ericsson/codechecker/blob/9004c1020fb9089c76b7454629b6d3fa96483e07/snap/snapcraft.yaml#L20
Main readme file also mentions Python 3.6: https://github.com/Ericsson/codechecker/blob/master/docs/README.md
All the github actions still uses Python 3.6

Try to grep on `3.6` in the source code and you will find more places.

I am not against it to upgrade the required Python version but we have to do it nicely and update all the other places as well. And also we haven't forget to mention it in the release notes because it is very important and by the way a backward compatibility change.

Reply:
Shouldn't the uplift to 3.7 be done in a separate PR?

Reply:
Fixit tag has been added for the future